prime-angular-frontend/src/app/modules/paper-enrolment/shared/guards/finalized-enrolment.guard.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable, Inject } from '@angular/core';
+import {
+  Router,
+  Params,
+  CanActivate,
+  ActivatedRouteSnapshot,
+  RouterStateSnapshot,
+  UrlTree,
+  UrlSegment
+} from '@angular/router';
+import { PaperEnrolmentResource } from '@paper-enrolment/services/paper-enrolment-resource.service';
+import { HttpEnrollee } from '@shared/models/enrolment.model';
+import { PaperEnrolmentRoutes } from '@paper-enrolment/paper-enrolment.routes';
+import { Observable } from 'rxjs';
+import { AppConfig, APP_CONFIG } from 'app/app-config.module';
+import { map } from 'rxjs/operators';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FinalizedEnrolmentGuard implements CanActivate {
+  constructor(
+    private router: Router,
+    @Inject(APP_CONFIG) private config: AppConfig,
+    private paperEnrolmentResource: PaperEnrolmentResource
+  ) { }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    const url = this.getUrl(state);
+    return this.checkAccess(url, next.params);
+  }
+
+  protected checkAccess(routePath: string = null, params: Params): Observable<boolean> | Promise<boolean> {
+    const enrolleeId = +params.eid;
+    if (enrolleeId) {
+      return this.paperEnrolmentResource.getEnrolleeById(enrolleeId)
+        .pipe(
+          map((enrollee: HttpEnrollee) => {
+            return this.routeDestination(routePath, enrollee, params);
+          })
+        );
+    } else {
+      return new Promise(async (resolve, reject) => resolve(true));
+    }
+  }
+
+  private routeDestination(routePath: string, httpEnrollee: HttpEnrollee, params) {
+    if (!httpEnrollee?.approvedDate) {
+      return true;
+    }
+    if (routePath.includes('demographic')) {
```

Comment:
Not sure of this section of code that replaces demographic.  Were you having an issue with the routing?

Reply:
/demographic was always getting appended to the end of routePath

Reply:
You'll want to find out the reason for that instead curing the symptom.  If it is always getting appended without knowing more I would assume that you're entering the enrolment since that's the default route.